src/odemis/acq/move.py

```diff
@@ -2262,3 +2755,164 @@ def _getCurrentAlignerPositionLabel(self) -> int:
+                    md_pos = pos.get("x", 0), pos.get("y", 0)
+                    comp.updateMetadata({
+                        model.MD_POS: md_pos,
+                        model.MD_STAGE_POSITION_RAW: pos_dep}
+                        )
+            except Exception as e:
+                logging.error("Failed to update %s with new position: %s", comp, e)
+
+        # update the SEM focus position when the stage is moved to compensate for linked behavior
+        # TODO: update the self.sem_eucentric_focus when the user manually focuses.
+        if self.pm.use_linked_sem_focus_compensation:
+            try:
+                ebeam_focus = model.getComponent(role="ebeam-focus")
+                # get the eucentric focus position from the metadata
+                self.sem_eucentric_focus = self._stage_bare.getMetadata()[model.MD_CALIB].get("SEM-Eucentric-Focus", 7.0e-3)
+                f = ebeam_focus.moveAbs({"z": self.sem_eucentric_focus})
+                f.result()
+            except Exception as e:
+                logging.error(f"Failed to update ebeam-focus with new position: 'z': {self.sem_eucentric_focus}, {e}")
+
+    def _updateSpeed(self, dep_speed):
+        """
+        update the speed VA based on the dependency's speed
+        """
+        # stage_speed = self.pm.to_sample_stage_from_stage_movement(dep_speed)
+        self.speed._set_value(dep_speed, force_write=True)
+
+    @isasync
+    def moveRel(self, shift: Dict[str, float], **kwargs) -> Future:
+        """
+        :param shift: The relative shift to be made
+        :param **kwargs: Mostly there to support "update" argument
+        """
+        # missing values are assumed to be zero
+        shift_stage = self.pm.from_sample_stage_to_stage_movement(shift)
+        logging.debug("converted relative move from %s to %s", shift, shift_stage)
+        return self._stage_bare.moveRel(shift_stage, **kwargs)
+
+    @isasync
+    def moveAbs(self, pos: Dict[str, float], **kwargs) -> Future:
+        """
+        :param pos: The absolute position to be moved to
+        :param **kwargs: Mostly there to support "update" argument
+        """
+
+        # if key is missing from pos, fill it with the current position
+        for key in self.axes.keys():
+            if key not in pos:
+                pos[key] = self.position.value[key]
+
+        # pos is a position, so absolute conversion
+        pos_stage = self.pm.from_sample_stage_to_stage_position(pos)
+        logging.debug("converted absolute move from %s to %s", pos, pos_stage)
+        return self._stage_bare.moveAbs(pos_stage, **kwargs)
+
+    @isasync
+    def moveRelChamberReferential(self, shift: Dict[str, float]) -> Future:
+        """Move the stage vertically in the chamber. This is non-blocking.
+        From OpenFIBSEM.
+        The desired input shift (x, z) is transformed to x, y, z axis components such that the
+        the stage moves in the vertical direction in the chamber. For TFS, the z-axis is
```

Comment:
Please be even more specific about the behaviour. Is the "input shift" in the sample stage coordinates? After the move, where will appear (in the FoV) the point originally at the center of the FoV? It will be shifted along X by `x`, but what about `y` and `z`? Does it just behave as if there was a "z" axis perpendicular to the sample? If so, it should be part of the standard `moveRel()`...
Also, in which posture will this move work? SEM imaging only? FIB milling?

Reply:
updated